packages/components/icon/src/utils/wrapPhosphorIcon.tsx

```diff
@@ -15,12 +15,16 @@ export function wrapPhosphorIcon(PhosphorIcon: PhosphorIcon) {
     isActive = false,
     color = isActive ? tokens.blue500 : tokens.gray900,
     size = 'medium',
+    testId,
     ...props
   }: GeneratedIconProps & { weight?: IconWeight }) => {
+    const commonProps = { ...props, 'data-test-id': testId };
     return (
       <Icon
-        {...props}
-        as={() => <PhosphorIcon {...props} color={color} size={sizes[size]} />}
+        {...commonProps}
```

Comment:
Shouldn't `Icon` take the `testId` prop?

Reply:
I had tried that, but didn't work, but looking deeper into it, seems that using a function on the `as` prop, breaks how the Icon and underling Box works.
It expects a React Element and would pass the props of the Icon and Box to it, but we are passing an arrow function that don't accept props.

So whatever we pass here to the Icon is being left out

Reply:
I'm taking a look into how to fix that